java/gradle.java/src/org/netbeans/modules/gradle/java/api/GradleJavaSourceSet.java

```diff
@@ -365,6 +366,33 @@ public boolean outputContains(File f) {
     public Set<File> getOutputClassDirs() {
         return outputClassDirs != null ? outputClassDirs : Collections.<File>emptySet();
     }
+    
+    /**
+     * Represents an unknown value. This is different from a value that is not present,
+     * i.e. an output directory for a language that is not used in the project.
+     * @since 1.19
+     */
+    public static final File UNKNOWN = new File("");
+    
+    /**
+     * Returns output directories for the given source type in the sourceset. Returns
+     * null, if the source type has no output directories. Returns UNKNOWN, if the 
+     * output location is not known.
+     * 
+     * @param srcType language type
+     * @return location or {@code null}.
+     * @since 1.19
+     */
+    public File getOutputClassDir(SourceType srcType) {
+        File f = outputs.get(srcType);
+        if (f == null) {
```

Comment:
this if seems redundant as UNKNOWN.equals(f) evaluates false it f == null
 so maybe

return UNKNOWN.equals(f) ? UNKNOWN : f;

will do?

Reply:
Fixed in 01cb17c54a